docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -82,6 +82,33 @@ Dagster provides out-of-the-box I/O managers for popular storage systems: AWS S3
 
 I/O managers are provided through the [resources](/concepts/resources) system which means you can supply different I/O managers for the same assets or ops in different situations. For example, you might use an in-memory I/O manager for unit-testing and an Amazon S3 I/O manager in production.
 
+
+### When to use I/O managers
+
+I/O managers are a powerful tool that can simplify your code and allow you to easily modify where your assets are stored. However, they are not required, nor are they the best option, in all scenarios.
+
+Some assets do not need an I/O manager. For example, you may write an asset that executes a Snowflake query that creates a new table based on the data of another table. This asset would depend on the existing table, but does not need to load that table in memory in order to execute the query:
+
+```python file=/tutorial/saving/no_io_assets_examples.py startafter=start_snowflake_example endbefore=end_snowflake_example
+@asset(deps=[orders])
+def returns():
+    conn = get_snowflake_connection()
+    conn.execute(
+        "CREATE TABLE returns AS SELECT * from orders WHERE status = 'RETURNED'"
+    )
+```
+
+Other reasons you may not want to use I/O managers:
+
+- You want to run a SQL query that creates or updates a table in a database (the above example).
+- Your pipeline manages I/O on its own by using other libraries/tools that write to storage.
+- You have unique and custom storage requirements for each asset.
+- Your assets won't fit in memory (for example, a database table with billions of rows).
+- You have an existing pipeline that manages its own I/O and want to run it with Dagster with minimal code changes.
+- You simply prefer to have the reading and writing code explicitly defined in each asset.
+
```

Comment:
Do you think its worth having a companion list of scenarios when I/O managers would be helpful?

Reply:
Perhaps - what would that list look like?

Reply:
off the top of my head 
- there is consistency with how you want to store multiple assets (in the same folder, in the same db schema following the same naming conventions)
- you want to be able to swap IO managers for testing, or other environments 
- you always want to load the upstream asset into memory in downstream assets 

maybe a good test we could mention to users is that if they are writing the same code at the top and bottom of each asset function to load the upstream dependencies and save the asset, then factoring that out to an I/O manager could be a good idea

Reply:
i like the idea of moving it up